src/parser/ParsedQuery.cpp

```diff
@@ -324,6 +326,27 @@ void ParsedQuery::merge(const ParsedQuery& p) {
   _rootGraphPattern.recomputeIds(&_numGraphPatterns);
 }
 
+// _____________________________________________________________________________
+const std::vector<Variable>& ParsedQuery::getVisibleVariables() {
+  return std::visit(&parsedQuery::ClauseBase::getVisibleVariables, _clause);
+}
+
+// _____________________________________________________________________________
+void ParsedQuery::registerVariablesVisibleInQueryBody(
+    const vector<Variable>& variables) {
+  for (const auto& var : variables) {
+    registerVariableVisibleInQueryBody(var);
+  }
+}
+
+// _____________________________________________________________________________
+void ParsedQuery::registerVariableVisibleInQueryBody(const Variable& variable) {
+  auto addVariable = [&variable](auto& clause) {
+    clause.addVisibleVariable(variable);
+  };
+  std::visit(addVariable, _clause);
+}
```

Comment:
```suggestion
  std::visit(&parsedQuery::ClauseBase::addvisibleVariable, _clause);
}
```
I thought I suggested this before, but it seems like I didn't.